### PR TITLE
makes registration register, with caveats

### PR DESCRIPTION
**Pull request by eaceaser**
This fixes registration up, mostly.

What it does

* Converts significantly touched javascript to typescript.
* Rewrites registration up.

What it doesn't do

* Actually reports success of registration
* reports a useful error message about why registration failed
* pulls token from localstorage for reuse (but i dont think the previous code did)
* handles usernames, as this is no longer included in the registration payload;
  we will have to query the server for that.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -5,11 +5,25 @@
...
     "plugins": [
         "prettier"
     ],
     "env": {
         "es6": true,
         "browser": true
+    },
+    "rules": {
+        "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx", ".tsx"] }],
+        "jsx-a11y/no-static-element-interactions": 1,
```
Comment (eaceaser) [inline .eslintrc:18]: these accessibility rules were reporting as errors and were pissing me off. we can deal with them later.

File: src/components/header.jsx [hunk 2]
```diff
@@ -70,15 +70,15 @@ const Header = () => (
           <Link className="header__navbar-item" to="/about">
             Learn
           </Link>
-          {username ? (
-            <LoggedInLink currentUser={username} logOut={logOut} />
+          {token ? (
+            <LoggedInLink currentUser="FIXME" logOut={logout} />
```
Comment (eaceaser) [inline src/components/header.jsx:74]: we need to rewire usernames, can do that as a separate PR.

File: src/components/display.jsx [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
 import { Mutation } from 'react-apollo';
 import { PASS } from '../graphql/mutations';
-import AuthContext from '../utils/AuthContext';
+import { AuthStoreConsumer } from '../stores/authStore';
```
Comment (ianwessen) [inline src/components/display.jsx:4]: Is this a typescript naming convention? Repo/Store? I'm not so familiar. Can you explain the name change and role of these pieces?

**Review by ianwessen — APPROVED** [review]
This is all looks pretty good, but I'm unfamilar with the Typescript syntax. Let's talk about some of these changes at Go club.

File: src/components/display.jsx (hunk 3 above)
Comment (eaceaser) [reply src/components/display.jsx:4]: i just made some shit up. usually a repo is responsible for reifying models from somewhere. since i’m wrapping the context here i wanted to give it a different name, so i went with store. honestly we could call it authcontext still. that might be better.
